### PR TITLE
Simple kbfs activity integration

### PR DESCRIPTION
**Pull request by chrisnojima**
This does a pretty simple integration for kbfs notifications. Due to some missing code on the kbfs side we reset the async-writing state to off when we haven't gotten an fs activity in awhile. That'll get better later

### CODE REVIEW

**Comment by chrisnojima** [comment]
https://github.com/keybase/client/pull/4750

File: protocol/bin/flow.js [hunk 1]
```diff
@@ -154,7 +154,9 @@ function capitalize (s) {
 function analyzeMessages (json, project) {
   // ui means an incoming rpc. simple regexp to filter this but it might break in the future if
   // the core side doesn't have a consisten naming convention. (must be case insensitive to pass correctly)
-  const isUIProtocol = ['notifyCtl'].indexOf(json.protocol) === -1 && !!json.protocol.match(/^notify.*|.*ui$/i)
+  const isUIProtocol = ['notifyCtl'].indexOf(json.protocol) === -1 &&
+    !!json.protocol.match(/^notify.*|.*ui$/i) &&
+    !json.protocol.match(/NotifyFSRequest/)
```
Comment (chrisnojima) [inline protocol/bin/flow.js]: Some inconsistencies here, expected (sadly)

File: shared/actions/favorite.js [hunk 2]
```diff
@@ -262,11 +266,52 @@ function _notify (state) {
   previousNotifyState = newNotifyState
 }
 
+function * _setupKBFSChangedHandler (): SagaGenerator<any, any> {
+  yield put((dispatch: Dispatch) => {
+    let temp = true
+    setInterval(() => {
```
Comment (chrisnojima) [inline shared/actions/favorite.js]: just flipping the bit short term

File: desktop/app/menu-bar.js [hunk 3]
```diff
@@ -53,18 +55,8 @@ export default function () {
...
-  })
-
-  ipcMain.on('showTrayRegular', () => {
-    iconType = 'regular'
-    updateIcon(false)
-  })
-
-  ipcMain.on('showTrayBadged', () => {
-    iconType = 'badged'
+  ipcMain.on('showTray', (event, type) => {
```
Comment (chrisnojima) [inline desktop/app/menu-bar.js]: parameterize this vs the individual messages

**Comment by chrisnojima** [comment]
@keybase/react-hackers this is ready for review

**Comment by chrisnojima** [comment]
@keybase/react-hackers bump

File: shared/actions/favorite.js [hunk 4]
```diff
@@ -262,11 +267,33 @@ function _notify (state) {
...
 
+function * _setupKBFSChangedHandler (): SagaGenerator<any, any> {
+  yield put((dispatch: Dispatch) => {
+    const debouncedKBFSStopped = _.debounce(() => {
+      const badgeAction: Action = badgeApp('kbfsUploading', false)
+      dispatch(badgeAction)
+      dispatch({type: Constants.kbfsStatusUpdated, payload: {isAsyncWriteHappening: false}})
+    }, 2000)
+
+    engine().setIncomingHandler('keybase.1.NotifyFS.FSSyncActivity', ({status}) => {
```
Comment (MarcoPolo) [inline shared/actions/favorite.js]: This might be a common pattern we'll use, so maybe next time we should clean up this pattern

Comment (chrisnojima) [inline shared/actions/favorite.js]: i'll do the yield call stuff now. let's wait on the larger pattern until we do more integrations and see

Comment (chrisnojima) [inline shared/actions/favorite.js]: oh i didn't do this cause this is in a dispatch cuase of all the callbacks. its going in as-is

**Pull request merged**